lcls_tools/common/data_analysis/fit/method_base.py

```diff
@@ -0,0 +1,144 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from matplotlib import pyplot as plt
+
+
+class MethodBase(ABC):
+    """
+    Base abstract class for all fit methods, which serves as the bare minimum
+    skeleton code needed. Should be used only as a parent class to all method
+    models.
+    ---------------------------
+    Arguments:
+    param_names: list (list all of param names that the model will contain)
+    param_guesses: np.ndarray (array that contains a guess as to what
+        each param value is organized in the same order as param_names)
+    param_bounds: np.ndarray (array that contains the lower
+        and upper bound on for acceptable values of each parameter)
+    """
+
+    param_names: list = None
```

Comment:
This design pattern is exactly the kind of thing pydantic is meant to simplify. Can we a simpler way to accomplish this?